items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala

```diff
@@ -92,18 +98,59 @@ class SierraItemUpdater(sierraSource: SierraSource)(
 
     } yield accessConditions
 
-  private def getAvailableDates: Option[List[AvailabilitySlot]] =
-    // call Prismic to get opening times
-    // generate list of dates on now + opening times
-    Some(
-      List(
-        AvailabilitySlot(
-          "2022-01-10T10:00:00+0000",
-          "2022-01-11T10:00:00+0000"
-        )
-      )
+  private def setAvailableDates(
+    item: DisplayItem
+  ): Future[DisplayItem] = {
+    // there is only ever one location per physicalItem and one accessCondition per location,
+    // but this may sometimes be empty if it could not be fetched.
+    val accessCondition = item.locations.headOption
+      .flatMap(_.accessConditions.headOption)
+
+    if (accessCondition.exists(_.isRequestable)) {
+      getAvailableDates(accessCondition.head)
+        .map(availableDates => item.copy(availableDates = Some(availableDates)))
+    } else {
+      Future.successful(item)
+    }
+  }
+
+  private def getAvailableDates(
+    accessCondition: DisplayAccessCondition
+  ): Future[List[AvailabilitySlot]] = {
+    val venueFromDisplayAccessCondition = Map(
+      "online-request" -> "library"
+      // other venue to be added as DisplayAccessMethod id -> content-api venue title
     )
 
+    val hourAtVenue = clock.timeInLondon().getHour
+    val leadTimeInDays = accessCondition.method.id match {
+      case "online-request" if hourAtVenue < 10  => 1
+      case "online-request" if hourAtVenue >= 10 => 2
+    }
+    def daysAwayFromNow(slot: AvailabilitySlot): Int = {
+      val closeTime = ZonedDateTime.parse(slot.to)
+      ChronoUnit.DAYS.between(clock.timeInLondon(), closeTime).toInt
```

Comment:
Nit: if we're defining this method inline, then it makes sense to close over the time (ie bind the variable from the outer scope rather than asking `clock` again)

```suggestion
    val timeAtVenue = clock.timeInLondon()
    val leadTimeInDays = accessCondition.method.id match {
      case "online-request" if timeAtVenue.getHour < 10  => 1
      case "online-request" if timeAtVenue.getHour >= 10 => 2
    }
    def daysAwayFromNow(slot: AvailabilitySlot): Int = {
      val closeTime = ZonedDateTime.parse(slot.to)
      ChronoUnit.DAYS.between(timeAtVenue, closeTime).toInt
```

Also (more importantly actually) - I think I might've been wrong about using `closeTime` here; should it be the opening time instead?

Reply:
I think close v open doesn't really matter, `ChronoUnit.DAYS.between` only looks at the day which is the same for open and close (these places are not open overnight)

Reply:
You and I know that - but how does the application? There's nothing in the model that tells us a slot only covers a day (the same reason that I suggested making this change to `dropWhile`). If we write this in a way that doesn't make assumptions about the data, then it means that someone reading it in future doesn't have to know that context, and also makes it robust to any potential changes in the data.

I think it needs to use the opening time, as that means the lead time is sufficient that the item will be available if the user turns up right at the start of the slot - does that sound right?

Reply:
Done and done :)